test/RibbonThetaVaultYearn.js

```diff
@@ -0,0 +1,3531 @@
+
+        this.oracle = await setupOracle(params.underlyingPricer, ownerSigner);
+
+        if (params.depositAsset === WETH_ADDRESS) {
+          const weth = this.assetContract.connect(counterpartySigner);
+          await weth.deposit({ value: this.premium });
+          await weth.approve(SWAP_ADDRESS, this.premium);
+          return;
+        }
+
+        if (params.mintConfig) {
+          await mintToken(
+            this.assetContract,
+            params.mintConfig.contractOwnerAddress,
+            counterpartySigner,
+            SWAP_ADDRESS,
+            this.premium
+          );
+          return;
+        }
+      });
+
+      it("reverts when not called with manager", async function () {
+        await expect(
+          this.vault.connect(userSigner).rollToNextOption()
+        ).to.be.revertedWith("Only manager");
+      });
+
+      it("reverts when delay not passed", async function () {
+        await this.vault
+          .connect(managerSigner)
+          .commitAndClose(this.optionTerms);
+
+        // will revert when trying to roll immediately
+        await expect(
+          this.vault.connect(managerSigner).rollToNextOption()
+        ).to.be.revertedWith("Cannot roll before delay");
+
+        time.increaseTo(
+          (await this.vault.nextOptionReadyAt()).sub(BigNumber.from("1"))
+        );
+
+        await expect(
+          this.vault.connect(managerSigner).rollToNextOption()
+        ).to.be.revertedWith("Cannot roll before delay");
+      });
+
+      it("mints oTokens and deposits collateral into vault", async function () {
+        let pricePerShare = BigNumber.from(
+          (await this.collateralContract.pricePerShare()).toString()
+        );
+        const lockedAmount = wdiv(
+          wmul(this.depositAmount, LOCKED_RATIO),
+          pricePerShare
+        ).sub(1);
+
+        // Note: available + locked is total but with 0.05% slippage incurred
+        const startMarginBalance = await this.collateralContract.balanceOf(
+          MARGIN_POOL
+        );
+
+        await this.vault
+          .connect(managerSigner)
+          .commitAndClose(this.optionTerms);
+
+        await time.increaseTo(
+          (await this.vault.nextOptionReadyAt()).toNumber() + 1
+        );
+
+        const res = this.vault.connect(managerSigner).rollToNextOption();
+
+        await expect(res).to.not.emit(this.vault, "CloseShort");
+
+        await expect(res)
+          .to.emit(this.vault, "OpenShort")
+          .withArgs(this.oTokenAddress, lockedAmount, manager);
+
+        assert.equal(
+          (await this.vault.lockedAmount()).toString(),
+          wmul(this.depositAmount, LOCKED_RATIO).sub(1).toString()
+        );
+
+        assert.equal(
+          (await this.vault.assetBalance()).toString(),
+          wmul(this.depositAmount, WITHDRAWAL_BUFFER).toString()
+        );
+
+        assert.equal(
+          (await this.collateralContract.balanceOf(MARGIN_POOL))
+            .sub(startMarginBalance)
```

Comment:
We also need to check that the ETH balance of the vault has been completely wrapped into yvWETH (the remaining 10% left)

Reply:
👍